homeassistant/components/yandex_transport/sensor.py

```diff
@@ -79,18 +79,19 @@ def update(self):
         transport_list = stop_metadata["Transport"]
         for transport in transport_list:
             route = transport["name"]
-            if self._routes and route not in self._routes:
-                # skip unnecessary route info
-                continue
-            if "Events" in transport["BriefSchedule"]:
-                for event in transport["BriefSchedule"]["Events"]:
-                    if "Estimated" in event:
-                        posix_time_next = int(event["Estimated"]["value"])
-                        if closer_time is None or closer_time > posix_time_next:
-                            closer_time = posix_time_next
-                        if route not in attrs:
-                            attrs[route] = []
-                        attrs[route].append(event["Estimated"]["text"])
+            for thread in transport["threads"]:
+                if self._routes and route not in self._routes:
+                    # skip unnecessary route info
+                    continue
+                if "Events" in thread["BriefSchedule"]:
```

Comment:
Please use guard clauses and stop loop iterations early with `continue`

```suggestion
                if "Events" not in thread["BriefSchedule"]:
                   continue
               
               for event in thread["BriefSchedule"]["Events"]:
                   if "Estimated" not in event:
                       continue
```